src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1297,24 +1264,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
         if (node.tag === HostRoot) {
           const root: FiberRoot = (node.stateNode: any);
           scheduleRoot(root, priorityLevel);
-          // Depending on the priority level, either perform work now or
-          // schedule a callback to perform work later.
-          switch (priorityLevel) {
-            case SynchronousPriority:
-              performWork(SynchronousPriority, null);
-              return;
-            case TaskPriority:
-              // TODO: If we're not already performing work, schedule a
-              // deferred callback.
-              return;
```

Comment:
This is the topic I brought up at lunch. I believe you're the one who suggested this to me, and why I added this comment :)

Seems correct to me that if we `setState` at Task priority, but we're not already performing work, we should schedule a callback and perform it later. The other option is to treat it as synchronous work, I suppose. We need to decide whether "task" priority should be treated as semantically different from sync priority for non-nested updates.